front/src/modules/simulationResult/components/ManchetteWithSpaceTimeChart/utils.ts

```diff
@@ -1,21 +1,65 @@
 import type { PathLevel, HoveredItem } from '@osrd-project/ui-charts';
 
 import { PATH_COLORS } from 'modules/simulationResult/consts';
-import type { TrainId } from 'reducers/osrdconf/types';
-import { formatEditoastTrainIdToTrainScheduleId } from 'utils/trainId';
+import type { OccurrenceId, TrainId } from 'reducers/osrdconf/types';
+import { getPacedTrainIdFromOccurrenceId, isOccurrence } from 'utils/trainId';
 
-/* eslint-disable import/prefer-default-export */
-export const getIdFromTrainPath = (trainPath: string): TrainId =>
-  // TODO : Adapt this to handle paced train in space time chart in issue https://github.com/OpenRailAssociation/osrd/issues/10613
-  formatEditoastTrainIdToTrainScheduleId(+trainPath.split('-')[0]);
-
-export const getPathStyle = (
+const getPathStyle = (
   hovered: HoveredItem | null,
   path: { color: string; id: string },
-  dragging: boolean
-): { color: string; level?: PathLevel } => {
+  dragging: boolean,
+  selectedPathId?: TrainId
+): {
+  color: string;
+  level?: PathLevel;
+  border?: {
+    offset: number;
+    color: string;
+    width?: number;
+    backgroundColor?: string;
+  };
+} => {
   if (hovered && 'pathId' in hovered.element && path.id === hovered?.element.pathId && !dragging) {
     return { color: PATH_COLORS.HOVERED_PATH, level: 1 };
   }
+  // Apply occurrence style if selectedPathId is an occurence from the same paced
+  if (selectedPathId) {
+    if (isOccurrence(selectedPathId)) {
+      // Selected occurrence
+      if (path.id === selectedPathId) {
+        return {
+          color: PATH_COLORS.SELECTED_OCCURRENCE_PATH,
+          level: 1,
+          border: {
+            offset: 3.5,
+            color: 'transparent',
+            backgroundColor: '#FAE6F6',
+          },
+        };
+      }
+      // Other occurrences from the same paced
+      if (
+        getPacedTrainIdFromOccurrenceId(path.id as OccurrenceId) ===
```

Comment:
Maybe you should check here that the path.id is an occurence to avoid the cast

Reply:
done